app/models/katello/root_repository.rb

```diff
@@ -275,6 +276,20 @@ def ensure_valid_upstream_authorization
       end
     end
 
+    def ensure_valid_auth_url_token
```

Comment:
There's quite a bit of behavior in this method - do you mind adding a unit test?